src/components/DataAccessLayer.ts

```diff
@@ -27,10 +28,84 @@ export const getStaticRawData = (): DataMCQuestion[] => {
 };
 
 export const getConvertedStaticData = (): MCQuestion[] => {
-  return UtilConversion.convertQuestions(questions);
+  // Validate questions
+  const allDataQs: DataMCQuestion[] = getStaticRawData();
+  return UtilConversion.convertQuestions(validateMCQuestions(allDataQs));
 };
 
 export const getAllQuestionsFromApi = async (): Promise<MCQuestion[]> => {
   const allQuizzes = await NetworkCalls.getQuiz();
+
   return UtilConversion.convertQuestions(allQuizzes);
 };
+
+// helpers functions with typeguards and console warns
+// IMPORTANT: This function is also used to filter out invalid tags.
+function validateMCQuestions(allDataQs: DataMCQuestion[]): DataMCQuestion[] {
+  NetworkGuard.isMCQuestionArray(allDataQs)
+    ? console.info(
+        "%cAll questions are valid",
+        "color: #00FF00",
+        "\nTotal Questions Validated:",
+        allDataQs.length,
+      )
+    : console.warn(
+        "%cWARNING: Some questions contains invalid structure",
+        "color: #FF0000",
+      );
+
+  let invalidTags = 0;
```

Comment:
We can probably have an object rather than mutable variables and possibly use something like reduce?

```ts

const initialStats = {
  invalidTags: 0,
  noTags: 0,
  invalidQs: 0,
  totalTags: 0,
};

const stats = (allDataQs as DataMCQuestion[]).reduce((acc, dataQs) => {
  // Check question structure
  if (!NetworkGuard.isMCQuestion(dataQs)) {
    return { ...acc, invalidQs: acc.invalidQs + 1 };
  }

  const { tags } = dataQs;

  if (!tags || (Array.isArray(tags) && !tags.length)) {
    return { ...acc, noTags: acc.noTags + 1 };
  }

  const totalTags = acc.totalTags + tags.length;

  if (!NetworkGuard.isAllTags(tags)) {
    const validTags = tags.filter((tag) => NetworkGuard.isTag(tag));
    const invalidTags = acc.invalidTags + tags.length - validTags.length;
    tags = validTags; // remove all invalid tags
    return { ...acc, invalidTags, totalTags };
  }

  return { ...acc, totalTags };
}, initialStats);

logQsWarnings(stats.invalidQs, allDataQs.length, stats.invalidTags, stats.totalTags, stats.noTags);

return allDataQs;

```

Reply:
Fixed.